src/AppBundle/Resources/public/js/ui.deckedit.js

```diff
@@ -88,10 +88,10 @@ ui.build_faction_selector = function build_faction_selector() {
 	
 	$('[data-filter=faction_selector]').empty();
 	if (app.deck.choices && app.deck.choices.length > 0){
-		$('[data-filter=faction_selector]').show();
 		for (var i = 0; i < app.deck.choices.length; i++){
 			var choice = app.deck.choices[i];
 			if (choice.faction_select) {
+				$('[data-filter=faction_selector]').show();

```

Comment:
Moved this down since now there is more than one kind of selector, I think it belongs more in here?